lib/lockfile.js

```diff
@@ -0,0 +1,56 @@
+const request = require('request-promise')
+const promiseRetry = require('promise-retry')
+const Log = require('gk-log')
+const dbs = require('../lib/dbs')
+
+const env = require('./env')
+module.exports = {
+  getNewLockfile
+}
+
+const max404Retries = 5
+
+async function getNewLockfile (packageJson, lock, isNpm) {
+  const logs = dbs.getLogsDb()
+
+  const log = Log({
+    logsDb: logs,
+    accountId: 'lockfile',
+    repoSlug: null,
+    context: 'lockfile'
+  })
+
+  const type = isNpm ? 'npm' : 'yarn'
+
+  return promiseRetry((retry, number) => {
+    /*
+      if we get a 404 here, log, and try again a few times.
+    */
+    return request({
+      uri: `${env.EXEC_SERVER_URL}`,
+      method: 'POST',
+      json: true,
+      body: {
+        type,
+        packageJson,
+        lock
+      }
+    })
+      .catch(error => {
+        if (error.statusCode === 404) {
+          if (number === max404Retries) {
```

Comment:
why retry on `404` specifically? I would think that a `404` is pretty definitive in terms of “this can not be processed” and I would think retrying on network errors like `lib/dbs.js` is what we rather want, but I might be missing something obvious.